File: DataBindingEntityJava/app/src/main/java/com/websarva/wings/android/databindingentityjava/ui/MainViewModel.java
package com.websarva.wings.android.databindingentityjava.ui;

import androidx.lifecycle.ViewModel;

import com.websarva.wings.android.databindingentityjava.data.model.Rectangle;

public class MainViewModel extends ViewModel {
	Rectangle _rectangle;

	public MainViewModel() {
		generateNewRectangle();
	}

	public void generateNewRectangle() {
		int height = (int) (Math.random() * 10) + 1;
		int width = (int) (Math.random() * 10) + 1;
		_rectangle = new Rectangle(height, width);
	}

	public Rectangle getRectangle() {
		return _rectangle;
	}
}

File: DataBindingEntityKotlin/app/src/main/java/com/websarva/wings/android/databindingentitykotlin/ui/MainViewModel.kt
package com.websarva.wings.android.databindingentitykotlin.ui

import androidx.lifecycle.ViewModel
import com.websarva.wings.android.databindingentitykotlin.data.model.Rectangle

class MainViewModel : ViewModel() {
	var rectangle: Rectangle

	init {
		rectangle = Rectangle(0, 0)
		generateNewRectangle()
	}

	fun generateNewRectangle() {
		val height = (Math.random() * 10).toInt() + 1
		val width = (Math.random() * 10).toInt() + 1
		rectangle = Rectangle(height, width)
	}
}
